CMakeLists.txt

```diff
@@ -0,0 +1,114 @@
+cmake_minimum_required (VERSION 3.2)
+project (xtest C)
+
+include(${OPTEE_TEST_SDK}/host_include/conf.cmake)
+
+################################################################################
+# Packages
+################################################################################
+find_package(Threads REQUIRED)
+if(NOT THREADS_FOUND)
+	message(FATAL_ERROR "Threads not found")
+endif()
+
+include(GNUInstallDirs)
+
+################################################################################
+# Compiler flags:
+#   We want to use the same flags in the entire optee_client git
+################################################################################
+add_compile_options (
+	-Wall -Wbad-function-cast -Wcast-align
+	-Werror-implicit-function-declaration -Wextra
+	-Wfloat-equal -Wformat-nonliteral -Wformat-security
+	-Wformat=2 -Winit-self -Wmissing-declarations
+	-Wmissing-format-attribute -Wmissing-include-dirs
+	-Wmissing-noreturn -Wmissing-prototypes -Wnested-externs
+	-Wpointer-arith -Wshadow -Wstrict-prototypes
+	-Wswitch-default -Wunsafe-loop-optimizations
+	-Wwrite-strings -Werror -fPIC
+ 	-Wno-missing-field-initializers
+)
+
+find_program(CCACHE_FOUND ccache)
+if(CCACHE_FOUND)
+	set_property(GLOBAL PROPERTY RULE_LAUNCH_COMPILE ccache)
+	set_property(GLOBAL PROPERTY RULE_LAUNCH_LINK ccache)
+endif(CCACHE_FOUND)
+
+set (SRC
+	host/xtest/adbg/src/adbg_case.c
+	host/xtest/adbg/src/adbg_enum.c
+	host/xtest/adbg/src/adbg_expect.c
+	host/xtest/adbg/src/adbg_log.c
+	host/xtest/adbg/src/adbg_run.c
+	host/xtest/adbg/src/security_utils_hex.c
+	host/xtest/aes_perf.c
+	host/xtest/benchmark_1000.c
+	host/xtest/benchmark_2000.c
+	host/xtest/regression_1000.c
+	host/xtest/regression_4000.c
+	host/xtest/regression_5000.c
+	host/xtest/regression_6000.c
+	host/xtest/regression_7000.c
+	host/xtest/regression_8000.c
+	host/xtest/sha_perf.c
+	host/xtest/xtest_helpers.c
+	host/xtest/xtest_main.c
+	host/xtest/xtest_test.c
+)
+
+if (CFG_GP_SOCKETS)
+	list (APPEND SRC
+		host/xtest/regression_2000.c
+		host/xtest/sock_server.c
+		host/xtest/rand_stream.c
+	)
+endif()
+
+if (CFG_SECSTOR_TA_MGMT_PTA)
+	list (APPEND SRC host/xtest/install_ta.c)
+endif()
+
+if (CFG_SECURE_DATA_PATH)
+	list (APPEND SRC host/xtest/sdp_basic.c)
+endif()
+
+################################################################################
+# Built binary
+################################################################################
+add_executable (${PROJECT_NAME} ${SRC})
+
+target_compile_options (${PROJECT_NAME} PRIVATE -include conf.h)
```

Comment:
Where do you pick up `conf.h` from? `./out/arm/include/generated/conf.h`? I cannot see that being mentioned in this patch.

Reply:
Ah, you are probably picking it up from `./out/arm/export-ta_arm32/host_include/conf.h` with the `PRIVATE ${OPTEE_TEST_SDK}/host_include`